sunpy/coordinates/frames.py

```diff
@@ -469,7 +469,13 @@ class Helioprojective(SunPyBaseCoordinateFrame):
 
     rsun = QuantityAttribute(default=_RSUN, unit=u.km)
     observer = ObserverCoordinateAttribute(HeliographicStonyhurst)
+    
+    @property
+    def angular_radius(self):
+        from sunpy.coordinates.sun import _angular_radius  # avoiding a circular import
+        return _angular_radius(self.rsun, self.observer.radius)
```

Comment:
```suggestion

        if not isinstance(self.observer, HeliographicStonyhurst):
            if self.observer is None:
                raise ValueError("The observer must be defined, not `None`.")
            raise ValueError("The observer must be fully defined by specifying `obstime`.")

        return _angular_radius(self.rsun, self.observer.radius)
```
Better error messages